packages/svelte/src/compiler/phases/2-analyze/index.js

```diff
@@ -268,12 +267,7 @@ export function analyze_component(root, options) {
 			!Runes.includes(name) ||
 			(declaration !== null &&
 				// const state = $state(0) is valid
-				!Runes.includes(
-					/** @type {string} */ (
-						get_callee_name(/** @type {import('estree').Expression} */ (declaration.initial))
-					)
-				) &&
-				// allow `import { derived } from 'svelte/store'` in the same file as `const x = $derived(..)` because one is not a subscription to the other
+				get_rune(declaration.initial, instance.scope) === null &&
```

Comment:
```suggestion
				get_rune(declaration.initial, instance.scope) === null &&
				// allow `import { derived } from 'svelte/store'` in the same file as `const x = $derived(..)` because one is not a subscription to the other
```